packages/query/src/block-processor.ts

```diff
@@ -361,45 +371,66 @@ export class BlockProcessor implements BlockProcessorInterface {
     return spentNullifiers;
   }
 
-  private async identifyLpNftPositions(txs: Transaction[]) {
```

Comment:
You'll probably want to review the rest of this file with "Hide whitespace" turned on. In short, I:

1. created a new `processTransactions` method that iterates over each transaction, then each action of each transaction, and calls `this.identifyAuctionNfts()` and `this.identifyLpNftPositions()` on each _action_.
2. created a new `identifyAuctioNfts()` method that saves auction NFT metadata when it appears.
3. modified `identifyLpNftPositions()` to process individual _actions_, rather than processing an array of transactions.

Reply:
this a great design choice as it's extensible to other metadata-related processing we'll need to do in the future!